virtual_rainforest/soil/model.py

```diff
@@ -0,0 +1,67 @@
+"""The `soil.model` module.
+
+The `soil.model` module creates a `SoilModel` class, which extended the base `Model`
+class to be usable to simulate the soil.
+"""
+
+from numpy import datetime64, timedelta64
+
+from virtual_rainforest.core.logger import LOGGER, log_and_raise
+from virtual_rainforest.core.model import BaseModel
+
+
+class InitialisationError(Exception):
+    """Custom exception class for model initialisation failures."""
+
+
+class SoilModel(BaseModel):
+    """A class describing the soil model.
+
+    Describes the specific functions and attributes that the soil module should possess.
+    It's very much incomplete at the moment, and only overwrites one function in a
+    pretty basic manner. This is intended as a demonstration of how the class
+    inheritance should be handled for the model classes.
+
+    Args:
+        start_time: Point in time that the model simulation should be started.
+        end_time: Time that the model simulation should end
+        update_interval: Time to wait between updates of the model state.
+
+    Attributes:
+        name: Names the model that is described
+    """
+
+    name = "soil"
+
+    def __init__(
+        self,
+        start_time: datetime64,
+        end_time: datetime64,
+        update_interval: timedelta64,
```

Comment:
Agree on `end_time`. In the case of `start_time` we need something to initially populate `last_update` so that the `should_update` function can run properly. We could use 0.0 and just decouple model timing from overall simulation (`datetime64`) timing. I think converting between times would be a fairly big source of potential errors though. So think we should keep `start_time` as an input, even it is just used to give `last_update` an initial value

Reply:
Interesting point. I guess I had been thinking that `should_update` is initialised as `True` so that it **always** runs first time (and with an order defined by the model order in the config?). So when the simulation starts up:

* the `spinup` method is run for each model 
* then we start the timing loop with a single common timeline 
* each model gets run for the first time in the first pass and then sets when next to run it

Otherwise, we could conceivably have a model that is badly configured not to run until after it is needed.